### PR TITLE
[CARBONDATA-3398] Handled show cache for index server and MV

### PR DESCRIPTION
**Pull request by kunal642**
1. Added support to show/drop metacahe information from index server.
2. Added tableNotFoundException fix when dbName and tableName have '_' in their names, while splitting using '_' the dbName was extracted wrongly. Instead now dbname and tableName would be seperated by '-' internally for show cache

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3370/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11637/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3371/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3574/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11638/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3374/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3577/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11641/

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java [hunk 1]
```diff
@@ -302,13 +302,14 @@ public synchronized void clear() {
...
 
-  @Override public String getCacheSize() throws IOException {
+  @Override public String getCacheSize() {
     long sum = 0L;
     int numOfIndexFiles = 0;
     for (Map.Entry<String, Set<TableBlockIndexUniqueIdentifier>> entry : segmentMap.entrySet()) {
       for (TableBlockIndexUniqueIdentifier tableBlockIndexUniqueIdentifier : entry.getValue()) {
-        sum += cache.get(new TableBlockIndexUniqueIdentifierWrapper(tableBlockIndexUniqueIdentifier,
-            getCarbonTable())).getMemorySize();
+        sum += cache.getIfPresent(
```
Comment (NamanRastogi) [inline core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java]: Put a null-check over `cache.getIfPresent(...)`

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3383/

**Comment by kunal642** [comment]
@ravipesala please review

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java (hunk 1 above)
Comment (kunal642) [reply core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java]: done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3586/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11650/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3410/

File: datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneFineGrainDataMapFactory.java [hunk 2]
```diff
@@ -112,4 +112,5 @@ public DataMapLevel getDataMapLevel() {
         return false;
     }
   }
+
```
Comment (ravipesala) [inline datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneFineGrainDataMapFactory.java]: Remove unnecessary change

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala [hunk 3]
```diff
@@ -559,4 +559,6 @@ class CGDataMapTestCase extends QueryTest with BeforeAndAfterAll {
         CarbonCommonConstants.ENABLE_QUERY_STATISTICS_DEFAULT)
   }
 
+
+
```
Comment (ravipesala) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: Don't change file if not needed

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala [hunk 4]
```diff
@@ -43,15 +44,25 @@ class DistributedShowCacheRDD(@transient private val ss: SparkSession, tableName
...
+                                     (tableName.nonEmpty && tableList.contains(table)) =>
         val sizeAndIndexLengths = tableDataMaps.asScala
-          .map(_.getBlockletDetailsFetcher.getCacheSize)
-        // return tableName_indexFileLength_indexCachesize for each executor.
-        sizeAndIndexLengths.map {
-          x => s"$table:$x"
-        }
+          .map { dataMap =>
+            if (!dataMap.getDataMapSchema.getProviderName
+              .equals(DataMapClassProvider.BLOOMFILTER.getShortName)) {
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala]: I don't get why we are specifically checking for `BLOOMFILTER`. We are not supposed to check anything specific

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3613/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11677/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3415/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3419/

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala (hunk 4 above)
Comment (kunal642) [reply integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala]: changed the condition to check if dataMapName is null (for BlockletDataMap)

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala (hunk 3 above)
Comment (kunal642) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: removed

File: datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneFineGrainDataMapFactory.java (hunk 2 above)
Comment (kunal642) [reply datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneFineGrainDataMapFactory.java]: reverted

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11682/

**Comment by ravipesala** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3420/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3618/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3623/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11686/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3628/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3428/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11691/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3429/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3430/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3633/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11697/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3431/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11698/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3634/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3435/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3638/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11702/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3437/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3438/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3439/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11706/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3642/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3440/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3441/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3644/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11708/

**Comment by kunal642** [comment]
@ravipesala Please review again.. There are some new changes in the PR

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3443/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3646/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11710/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3446/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3649/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11713/

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala [hunk 5]
```diff
@@ -43,15 +50,23 @@ class DistributedShowCacheRDD(@transient private val ss: SparkSession, tableName
...
+      case (table, tableDataMaps) if tableName.isEmpty || tableList.contains(table) =>
         val sizeAndIndexLengths = tableDataMaps.asScala
-          .map(_.getBlockletDetailsFetcher.getCacheSize)
-        // return tableName_indexFileLength_indexCachesize for each executor.
-        sizeAndIndexLengths.map {
-          x => s"$table:$x"
-        }
+          .map { dataMap =>
+            // if datamap name is null then it means that the TableDataMap is of BlockletDatamap
+            // type.
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala]: This statement seems not valid, update the comment

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 6]
```diff
@@ -62,216 +68,397 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
+        case Nil => ("", 0, 0L, "")
+      }
+      val parentDictionary = getDictionarySize(carbonTable)(sparkSession)
+      val childMetaCacheInfos = childTableList.flatMap {
+        childTable =>
+          val tableArray = childTable._1.split("-")
+          val dbName = tableArray(0)
+          val tableName = tableArray(1)
+          if (childTable._2
+            .equalsIgnoreCase(DataMapClassProvider.BLOOMFILTER.getShortName)) {
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Please don't check the specific datamap type anywhere, we don't want to update this code every time new datamap is added.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 7]
```diff
@@ -62,216 +68,397 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
+    var dbIndexSize = 0L
+    cache.getCacheMap.asScala.foreach {
+      case (key, cacheable) =>
+        cacheable match {
+          case _: BlockletDataMapIndexWrapper =>
+            allIndexSize += cacheable.getMemorySize
+            if (tablePaths.exists { path => key.startsWith(path) }) {
+              dbIndexSize += cacheable.getMemorySize
+            }
+          case _: BloomCacheKeyValue.CacheValue =>
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Please don't use specific datamap

Comment (kunal642) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: changed

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 6 above)
Comment (kunal642) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: removed the specific check. There are 2 places where this kind of check still exists in case of index server because i dont have the TableDataMap object to differentiate between Bloom and others. Index server only returns the provider name

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3497/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11767/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3701/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3501/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3706/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11772/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3505/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3709/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11775/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3507/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11778/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3712/

**Comment by kunal642** [comment]
@ravipesala build passed

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 6 above)
Comment (ravipesala) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: I am not sure why we need to know about Bloom type. We just want to display the cache size occupied in memory for each datamap. Why we need a specific bloom check?

**Comment by ravipesala** [comment]
LGTM

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java [hunk 8]
```diff
@@ -302,14 +302,18 @@ public synchronized void clear() {
     }
   }
 
-  @Override public String getCacheSize() throws IOException {
+  @Override public String getCacheSize() {
```
Comment (xubo245) [inline core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java]: Please optimize this line to two line:
`
  @Override
  public String getCacheSize() {
`

File: datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMapFactory.java [hunk 9]
```diff
@@ -453,4 +453,19 @@ public DataMapMeta getMeta() {
   public DataMapLevel getDataMapLevel() {
     return DataMapLevel.CG;
   }
+
+  @Override public String getCacheSize() {
```
Comment (xubo245) [inline datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMapFactory.java]: Please optimize

File: hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java [hunk 10]
```diff
@@ -565,7 +565,9 @@ public BlockMappingVO getBlockRowCount(Job job, CarbonTable table,
                   allSegments.getInvalidSegments(), toBeCleanedSegments));
           for (InputSplit extendedBlocklet : extendedBlocklets) {
             CarbonInputSplit blocklet = (CarbonInputSplit) extendedBlocklet;
-            blockletToRowCountMap.put(blocklet.getSegmentId() + "," + blocklet.getFilePath(),
+            String filePath = blocklet.getFilePath();
+            String blockName = filePath.substring(filePath.lastIndexOf("/") + 1);
```
Comment (xubo245) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java]: Whether have "\\" in  filePath? like for windows?

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala [hunk 11]
```diff
@@ -47,7 +47,7 @@ trait ServerInterface {
   /**
    * Get the cache size for the specified table.
    */
-  def showCache(tableName: String) : Array[String]
+  def showCache(tableNames: String) : Array[String]
```
Comment (xubo245) [inline integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala]: Please optimize the description.  table=> tables

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 12]
```diff
@@ -62,216 +67,366 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
+    val childTables = getChildTableList(mainTable)(sparkSession)
+    val cache = if (tableIdentifier.nonEmpty) {
+      executeJobToGetCache(childTables.map(_._1) ++ List(mainTable.getTableUniqueName))
+    } else {
+      cacheResult
+    }
+    val (mainTableFiles, mainTableCache) = getTableCache(cache, mainTable.getTableUniqueName)
+    val childMetaCacheInfos = childTables.flatMap {
+      childTable =>
+        val tableName = childTable._1.replace("-", "_")
```
Comment (xubo245) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Why do you need replace tableName?

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala [hunk 13]
```diff
@@ -43,15 +49,21 @@ class DistributedShowCacheRDD(@transient private val ss: SparkSession, tableName
 
   override def internalCompute(split: Partition, context: TaskContext): Iterator[String] = {
     val dataMaps = DataMapStoreManager.getInstance().getAllDataMaps.asScala
+    val tableList = tableName.split(",").map(_.replace("-", "_"))
```
Comment (xubo245) [inline integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala]: Why do you need replace tableName?

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java (hunk 8 above)
Comment (kunal642) [reply core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java]: done

File: datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMapFactory.java (hunk 9 above)
Comment (kunal642) [reply datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMapFactory.java]: done

File: hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java (hunk 10 above)
Comment (kunal642) [reply hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java]: added replace("\\", "/") to handle for windows

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala (hunk 11 above)
Comment (kunal642) [reply integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 12 above)
Comment (kunal642) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: because the table unique names are in the format dbName-tableName to ease the parsing for dbName and tableName.

If i use _ and dbName and tableName also have _ in their names then the split will become difficult.

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala (hunk 13 above)
Comment (kunal642) [reply integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala]: same as above

**Comment by xubo245** [comment]
LGTM if CI ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3667/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11940/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3878/
